lib/queue.rb

```diff
@@ -1,31 +1,57 @@
 class Queue
-
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+    end
+
+    if @front == @back
+      raise Error, "Queue is full!"
+    end
+
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
+
+    if @store[@front].nil?
+      @front = (@front + 1) % @store.length
+    end
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+    end
+    if @front == @back
+      raise Error.new "Queue is empty!"
+    end
+    element = @store[@front]
+    @front = (@front + 1) % @store.length
+    return element
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    @store.length
```

Comment:
This only gives you the size of store, not the number of items being used by the queue.

Better something like this:

```suggestion
    return @back - @front if @front < @back
    return @back + @store.length - @front
```